assets/js/data.coffee

```diff
@@ -278,5 +278,45 @@ class Data
 
     @loadTo serialized
 
+  export: (filename, mimetype) ->
+    jsonContent = do @serialize
+    filename ||= 'vimflowy.json'
+    if not mimetype? # Infer mimetype from file extension
+        extension = (filename.split ".")[1] ? 'txt'
+        extensionLookup =
+            'json': 'application/json'
+            'txt': 'text/plain'
+        mimetype = extensionLookup[extension.toLowerCase()]
+    if mimetype == 'application/json'
+        delete jsonContent.viewRoot
+        content = JSON.stringify(jsonContent, undefined, 2)
+    else if mimetype == 'text/plain'
+        # Workflowy compatible plaintext export
+        #   Ignores 'collapsed' and viewRoot
+        indent = "  "
+        exportLines = (node) ->
+            if typeof(node) == 'string'
+              return ["- #{node}"]
+            lines = []
+            lines.push "- #{node.line}"
+            for child in node.children ? []
+                for line in exportLines child
+                    lines.push "#{indent}#{line}"
+            return lines
+        content = (line for line in exportLines jsonContent).join "\n"
+    else
+        throw "Invalid export format"
+    @saveFile filename, mimetype, content
+
+  saveFile: (filename, mimetype, content) ->
+    if not $?
+      return content # Tests are running
```

Comment:
rather than this, make a function which returns the raw content (basically export, minus the last line), and unit test that.  

have this function call that function, rather than the other way around